app/uk/gov/hmrc/agentsubscriptionfrontend/service/SubscriptionService.scala

```diff
@@ -61,4 +82,50 @@ class SubscriptionService @Inject()(agentSubscriptionConnector: AgentSubscriptio
       case e => throw e
     }
   }
+
+  def completePartialSubscription(details: CompletePartialSubscriptionBody)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[(Boolean, Option[Arn])] =
+    for {
+      subscriptionProgress <- getSubscriptionStatus(details.utr, details.knownFacts.postcode) //FixMe APB-2805 this check should happen previously and can be stored in the session
```

Comment:
I think we shouldn't mix getSubscriptionStatus and completePartialSubscription. Identifying if the process is opted for partial subscription should be an independent in if you look at checkAgencyStatusGivenValidForm method in CheckAgencyController, mixing these two methods result in calling getSubscriptionStatus twice for no reason.

Reply:
https://jira.tools.tax.service.gov.uk/browse/APB-2805
will resolve this and the extra call in returnAfterGGCredsCreated